Remotion/Web/Core/Globalization/ResourceManagerExtensions.cs

```diff
@@ -0,0 +1,107 @@
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+using System;
+using Remotion.Globalization;
+using Remotion.Utilities;
+
+namespace Remotion.Web.Globalization
+{
+  public static class ResourceManagerExtensions
+  {
+    /// <summary>
+    ///   Gets the value of the specified String resource.
+    /// </summary>
+    /// <returns>
+    ///   The value of the resource as <see cref="WebString"/>. If no match is possible, the identifier is returned.
+    /// </returns>
+    public static WebString GetWebString (this IResourceManager resourceManager, string id, WebStringType webStringType)
+    {
+      ArgumentUtility.CheckNotNull ("resourceManager", resourceManager);
+      ArgumentUtility.CheckNotNull ("id", id);
+
+      var resourceString = resourceManager.GetString (id);
+
+      return CreateWebStringOfType (webStringType, resourceString);
+    }
+
+    /// <summary>
+    ///   Gets the value of the specified String resource.
+    /// </summary>
+    /// <returns>
+    ///   The value of the resource as <see cref="WebString"/>. If no match is possible, <see langword="null"/> is returned.
+    /// </returns>
+    public static WebString? GetWebStringOrDefault (this IResourceManager resourceManager, string id, WebStringType webStringType)
+    {
+      ArgumentUtility.CheckNotNull ("resourceManager", resourceManager);
+      ArgumentUtility.CheckNotNull ("id", id);
+
+      var resourceString = resourceManager.GetStringOrDefault (id);
+
+      if (resourceString == null)
+        return null;
+
+      return CreateWebStringOfType (webStringType, resourceString);
+    }
+
+    /// <summary>
+    ///   Gets the value of the specified string resource. The resource is identified by
+    ///   concatenating type and value name.
+    /// </summary>
+    /// <remarks> See <see cref="ResourceIdentifiersAttribute.GetResourceIdentifier"/> for resource identifier syntax. </remarks>
+    /// <returns>
+    ///   The value of the resource as <see cref="WebString"/>. If no match is possible, the identifier is returned.
+    /// </returns>
+    public static WebString GetWebString (this IResourceManager resourceManager, Enum enumValue, WebStringType webStringType)
+    {
+      ArgumentUtility.CheckNotNull ("resourceManager", resourceManager);
+      ArgumentUtility.CheckNotNull ("enumValue", enumValue);
+
+      var resourceString = resourceManager.GetString (enumValue);
+
+      return CreateWebStringOfType (webStringType, resourceString);
+    }
+
+    /// <summary>
+    ///   Gets the value of the specified string resource. The resource is identified by
+    ///   concatenating type and value name.
+    /// </summary>
+    /// <remarks> See <see cref="ResourceIdentifiersAttribute.GetResourceIdentifier"/> for resource identifier syntax. </remarks>
+    /// <returns>
+    ///   The value of the resource as <see cref="WebString"/>. If no match is possible, null is returned.
```

Comment:
please use langword
(yes, this is missing in the source docs. Same goes for the param-not-null docs)